File: neetcode/blind75/bit_manipulation/number_of_one_bits.py
# https://neetcode.io/problems/number-of-one-bits

class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while (n != 0):
            if (n % 2 == 1):
                count += 1

            n = math.floor(n/2)

        return count

File: neetcode/blind75_JS_2/Bit-Manipulation/number-of-one-bits.js
// https://neetcode.io/problems/number-of-one-bits

/*
while converting to binary rep, count the 1 bits

bit = currN (remainder op %) 2
currN = currN (floor /) 2
*/

class Solution {
    /**
     * @param {number} n - a positive integer
     * @return {number}
     */
    hammingWeight(n) {
        if (n === 0) {
            return 0
        }

        let count = 0
        let binaryRep = ''

        while (n !== 0) {
            let bit = n % 2
            if (bit === 1) {
                count += 1
            }

            binaryRep += bit.toString()

            n = Math.floor(n / 2)
        }

        console.log(binaryRep.reverse)
        return count
    }
}
